core/unit_test/TestTeamVector.hpp

```diff
@@ -109,7 +109,7 @@ struct functor_team_for {
           value += values(i);
         }
 
-        if (test != value) {
+        if (static_cast<double>(test) != static_cast<double>(value)) {
           printf("FAILED team_parallel_for %i %i %f %f\n", team.league_rank(),
```

Comment:
doesn't this need to "%lf" in the print?

Reply:
why do you need to change this at all?

Reply:
I'll fix the printf formatting like you mentioned, missed this spot.
@crtrott I made of the comparison changes to use the `static_assert` to address errors triggered with -Werror like "error: use of overloaded operator '!=' is ambiguous". I cast to double as a pattern match of the casting used for printing the same variables (test and value) in the line below, but will change the cast or resolve differently if you prefer this handled in a different way